sources/api/bootstrap-commands/src/main.rs

```diff
@@ -0,0 +1,345 @@
+            match args {
+                Ok(arguments) => {
+                    if let Err(_e) = command(API_CLIENT_PATH, arguments) {
+                        if essential {
+                            error::BootstrapCommandFailureSnafu { name };
+                        }
+                    }
+                },
+                _ => {
+                    if essential {
+                        error::BootstrapCommandFailureSnafu { name };
+                    }
+                },
+            }
+        }
+
+        if mode == "once" {
+            let formatted = format!("settings.bootstrap-commands.{}.mode=off", name);
+            info!("Turning off bootstrap-command '{}'", name);
+            command(API_CLIENT_PATH, ["set", formatted.as_str()])?;
+        }
+
+        info!("Succesfully ran bootstrap command '{}'", name);
+    }
+
+    Ok(())
+}
+
+/// Query the API for the currently defined bootstrap commands
+fn get_bootstrap_commands<P>(config_path: P) -> Result<HashMap<Identifier, BootstrapCommand>>
+where
+    P: AsRef<Path>,
+{
+    let config_str = fs::read_to_string(config_path.as_ref()).context(error::ReadConfigSnafu {
+        config_path: config_path.as_ref(),
+    })?;
+
+    let config: BootstrapCommandConfig =
+        toml::from_str(&config_str).context(error::DeserializationSnafu {
+            config_path: config_path.as_ref(),
+        })?;
+
+    Ok(config.bootstrap_commands.unwrap_or_default())
+}
+
+/// Parse the args to the program and return an Args struct
+fn parse_args(args: env::Args) -> Result<Args> {
+    let mut global_args = Args::default();
+
+    let mut iter = args.skip(1);
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            // Global args
+            "--log-level" => {
+                let log_level = iter.next().context(error::UsageSnafu {
+                    message: "Did not give argument to --log-level",
+                })?;
+                global_args.log_level = LevelFilter::from_str(&log_level)
+                    .context(error::LogLevelSnafu { log_level })?;
+            }
+
+            "-c" | "--config-path" => {
+                let config_str = iter.next().context(error::UsageSnafu {
+                    message: "Did not give argument to --config-path",
+                })?;
+                global_args.config_path = PathBuf::from(config_str.as_str());
+            }
+
+            _ => (),
+        }
+    }
+
+    Ok(global_args)
+}
+
+fn run() -> Result<()> {
+    let args = parse_args(env::args())?;
+
+    // SimpleLogger will send errors to stderr and anything less to stdout.
+    SimpleLogger::init(args.log_level, LogConfig::default()).context(error::LoggerSnafu)?;
+
+    let bootstrap_commands = get_bootstrap_commands(args.config_path)?;
+
+    // We want to run bootstrap commands in sorted order of their names.
+    for name in bootstrap_commands.keys().sorted() {
+        let bootstrap_command = &bootstrap_commands[name];
```

Comment:
I wonder if we should turn on the same clippy rule we used in the Bottlerocket Settings SDK, and avoid accessing elements in a `HashMap` like this, even though it is safe since you are getting the keys from the actual hash_map.

Reply:
I couldn't find a better way to get key, value where keys are in sorted order.

Reply:
Updated HashMap to BTreeMap which stores keys in BST (sorted order). Verified it works with end to end testing.